config/src/utils.rs

```diff
@@ -168,6 +169,38 @@ pub(crate) fn get_dir_remapping(dir: impl AsRef<Path>) -> Option<Remapping> {
     }
 }
 
+/// Returns all available `profile` keys in a given `.toml` file
+///
+/// i.e. The toml below would return would return `["default", "ci", "local"]`
+/// ```toml
+/// [profile.default]
+/// ...
+/// [profile.ci]
+/// ...
+/// [profile.local]
+/// ```
+pub fn get_available_profiles(toml_path: impl AsRef<Path>) -> eyre::Result<Vec<String>> {
+    let doc = toml(toml_path)?;
+    let mut result = vec![Config::DEFAULT_PROFILE.to_string()];
+
+    if let Some(Item::Table(profiles)) = doc.as_table().get(Config::PROFILE_SECTION) {
+        for (_, (profile, _)) in profiles.iter().enumerate() {
+            let p = profile.to_string();
+            if !result.contains(&p) {
+                result.push(p);
+            }
+        }
+    }
+
+    Ok(result)
+}
+
+fn toml(path: impl AsRef<Path>) -> eyre::Result<Document> {
```

Comment:
Can we add some docs for this function, even if it's used internally? Also, suggesting another name so it's a bit clearer on what it does if you don't read the function body:
```suggestion
fn read_toml(path: impl AsRef<Path>) -> eyre::Result<Document> {
```